src/Oracle/Oracle.cs

```diff
@@ -0,0 +1,368 @@
+                    while (enumerator.MoveNext() && !CancelSource.IsCancellationRequested)
+                    {
+                        if (PendingQueue.TryGetValue(enumerator.Current.RequestId, out OracleTask task) && task.Tx != null)
+                            continue;
+                        ProcessRequest(snapshot, enumerator.Current.RequestId, enumerator.Current.Request);
+                    }
+                    Thread.Sleep(500);
+                }
+            }).Start();
+        }
+
+        private void CheckOracleAccount(StoreView snapshot)
+        {
+            var oraclePubs = NativeContract.Oracle.GetOracleNodes(snapshot);
+            if (oraclePubs.Length == 0) throw new ArgumentException("The oracle service is unavailable");
+            var oracles = oraclePubs.Select(u => Contract.CreateSignatureRedeemScript(u).ToScriptHash());
+            var accounts = Wallet?.GetAccounts()
+                .Where(u => u.HasKey && !u.Lock && oracles.Contains(u.ScriptHash))
+                .Select(u => (u.Contract, u.GetKey()))
+                .ToArray();
+            if (accounts.Length == 0) throw new ArgumentException("The wallet doesn't have any oracle accounts");
+        }
+
+        [ConsoleCommand("stop oracle", Category = "Oracle", Description = "Stop oracle service")]
+        private void OnStop()
+        {
+            Interlocked.Exchange(ref CancelSource, null)?.Cancel();
+        }
+
+        public void SendResponseSignature(ulong requestId, byte[] txSign, KeyPair keyPair)
+        {
+            var message = keyPair.PublicKey.ToArray().Concat(BitConverter.GetBytes(requestId)).Concat(txSign).ToArray();
+            var sign = Crypto.Sign(message, keyPair.PrivateKey, keyPair.PublicKey.EncodePoint(false)[1..]);
+            var content = "{ \"id\": " + (++Counter) + ", \"jsonrpc\": \"2.0\",  \"method\": \"submitoracleresponse\",  \"params\":[\"" + message.Concat(sign).ToArray().ToHexString() + "\"] }";
+
+            foreach (var node in Nodes)
+            {
+                new Task(() =>
+                {
+                    try
+                    {
+                        var url = new Uri(node);
+                        HttpWebRequest request = (HttpWebRequest)WebRequest.Create(url);
+                        request.Method = "POST";
+                        request.ContentType = "application/json";
+                        using (StreamWriter dataStream = new StreamWriter(request.GetRequestStream()))
+                        {
+                            dataStream.Write(content);
+                            dataStream.Close();
+                        }
+                        HttpWebResponse response = (HttpWebResponse)request.GetResponse();
+                        StreamReader reader = new StreamReader(response.GetResponseStream(), Encoding.GetEncoding("UTF-8"));
```

Comment:
Use rpc client?

Reply:
We should force to has a `Content-Length` header and ensure that the return it's less than X (policy value).

Reply:
Done

Reply:
I think that the fix it's wrong, we need to check the `ContentLenght` in the response

Reply:
We should be able to configure the max allowed size for responses